src/encrypt.rs

```diff
@@ -24,13 +24,16 @@ fn same_mask_u8(x: u16, y: u16) -> u8 {
 /// Does not take any input arguments.
 /// If generation of pseudo-random numbers fails, an error is returned.
 #[cfg(not(any(feature = "mceliece8192128", feature = "mceliece8192128f")))]
-fn gen_e(e: &mut [u8; SYS_N / 8], rng: &mut impl RNGState) -> Result<(), Box<dyn error::Error>> {
+fn gen_e<R: CryptoRng + RngCore>(
+    e: &mut [u8; SYS_N / 8],
+    rng: &mut R,
+) -> Result<(), Box<dyn error::Error>> {
     let mut ind = [0u16; SYS_T];
     let mut val = [0u8; SYS_T];
 
     loop {
         let mut bytes = [0u8; SYS_T * 4];
-        rng.randombytes(&mut bytes)?;
+        rng.fill_bytes(&mut bytes);
```

Comment:
Can we just ignore the possibility of fallible rngs here? There is also `RngCore::try_fill_bytes` we could use to propagate any error to the caller rather than panicking.

In practice most people will probably use this library with some rng automatically seeded from `OsRng`, and `OsRng` can fail, albeit [very unlikely](https://github.com/rust-random/rand/blob/25821152903d4bb1182d2e282819530441411f32/rand_core/src/os.rs#L25-L34).

If this crate is soon to adopt the RustCrypto `kem` traits, those are all fallible anyway. That means the API of this crate will be fallible anyway. So we try to handle the rng errors nicely, if we are going to pay the fallibility complexity in the API no matter what.

Reply:
I removed them because in #9  I also removed all of the boxed errors to allow for `no_std`. I guess since the KEM API is fallible it makes sense to support this and use `try_fill_bytes` everywhere. I would do that in a later PR though, when moving to the KEM API, since the error handling probably needs some more consideration there.